File: source/aoBlogs2/Models/BlogImageModel.vb

Imports Contensive.BaseClasses
Imports Contensive.Models.Db

Namespace Models
    Public Class BlogImageModel
        Inherits DbBaseModel
        '
        '====================================================================================================
        ''' <summary>
        '''table definition
        '''</summary>
        Public Shared ReadOnly Property tableMetadata As DbBaseTableMetadataModel = New DbBaseTableMetadataModel("Blog Images", "BlogImages", "default", False)
        '
        '====================================================================================================
        ' -- instance properties
        '
        Public Property AltSizeList As String
        Public Property description As String
        Public Property Filename As DbBaseModel.FieldTypeFile
        Public Property height As Integer
        Public Property width As Integer
        '
        ''' <summary>
        ''' Return a list of blog entry images for the blog entry
        ''' </summary>
        ''' <param name="cp"></param>
        ''' <param name="entryId">The id of the blog entry</param>
        ''' <returns></returns>
        Public Shared Function createListFromBlogEntry(cp As CPBaseClass, entryId As Integer) As List(Of BlogImageModel)
            Try
                Dim result As New List(Of BlogImageModel)
                For Each Rule In DbBaseModel.createList(Of BlogImageRuleModel)(cp, "(BlogEntryID=" & entryId & ")")
                    Dim blogimage As BlogImageModel = DbBaseModel.create(Of BlogImageModel)(cp, Rule.BlogImageID)
                    If (blogimage IsNot Nothing) Then result.Add(blogimage)
                Next
                Return result
            Catch ex As Exception
                cp.Site.ErrorReport(ex)
                Throw
            End Try
        End Function
    End Class
End Namespace

File: source/aoBlogs2/Models/Db/BlogImageModel.cs

using Contensive.BaseClasses;
using Contensive.Models.Db;
using System;
using System.Collections.Generic;
using System.Data;

namespace Contensive.Blog.Models {
    public class BlogImageModel : Contensive.Models.Db.DbBaseModel {        // <------ set set model Name and everywhere that matches this string
        // 
        // ====================================================================================================
        public static DbBaseTableMetadataModel tableMetadata { get; private set; } = new DbBaseTableMetadataModel("Blog Images", "BlogImages", "default", false);
        // 
        // ====================================================================================================
        // -- instance properties
        // instancePropertiesGoHere
        public string altSizeList { get; set; }
        public string description { get; set; }
        public string Filename { get; set; }
        public int height { get; set; }
        public int width { get; set; }
        /// <summary>
        /// for images attached directly to the post. blank for images shared across posts
        /// </summary>
        public int blogEntryId { get; set; }

        public string getUploadPath(string fieldName) {
            return tableMetadata.tableNameLower + "/" + fieldName.ToLower() + "/" + id.ToString().PadLeft(12, '0') + "/";
        }
        // 
        /// <summary>
        /// Return a list of blog entry images for the blog entry
        /// </summary>
        /// <param name="cp"></param>
        /// <param name="entryId">The id of the blog entry</param>
        /// <returns></returns>
        public static List<BlogImageModel> createListFromBlogEntry(CPBaseClass cp, int entryId) {
            var result = new List<BlogImageModel>();
            try {
                string sql = $@"
                    select distinct
                        i.*
                    from
	                    BlogImages i
	                    left join BlogImageRules r on r.blogimageid=i.id
                    where
	                    i.blogentryid={entryId}
	                    or (r.blogentryid={entryId})
                    order by
                        i.sortOrder, i.id
                    ";
                // create a list of blogimagemodel records for all images referenced directly and indirectly by the blog image rules
                using (DataTable dt = cp.Db.ExecuteQuery(sql.Replace("{entryId}", entryId.ToString()))) {
                    foreach (DataRow dr in dt.Rows) {
                        var blogimage = new BlogImageModel();
                        blogimage.load<BlogImageModel>(cp, dr);
                        result.Add(blogimage);
                    }
                }
            } catch (Exception ex) {
                cp.Site.ErrorReport(ex);
            }
            return result;
        }
    }
}